src/passes/RemoveUnusedModuleElements.cpp

```diff
@@ -62,15 +62,16 @@ struct ReachabilityAnalyzer : public PostWalker<ReachabilityAnalyzer> {
       queue.pop_back();
       if (reachable.count(curr) == 0) {
         reachable.insert(curr);
-        if (curr.first == ModuleElementKind::Function) {
+        auto [kind, value] = curr;
```

Comment:
```suggestion
        auto& [kind, value] = curr;
```
Also, isn't `curr` a dangling reference after the `pop_back` call? The `pop_back` should probably be moved to happen after we are done with `curr`.

Reply:
You mean move it like this?
before:
```cpp
while (queue.size()) {
  auto& curr = queue.back();
  queue.pop_back();
  if (reachable.count(curr) == 0) {
     reachable.insert(curr);
     auto& [kind, value] = curr;
     ....
  }
}
```
to
```cpp
while (queue.size()) {
  auto& curr = queue.back();
  if (reachable.count(curr) == 0) {
     reachable.insert(curr);
     auto& [kind, value] = curr;
     ....
  }
  queue.pop_back();
}
```